File: Selenium/Activities/Activity8.py
# Set up the Firefox Driver with WebDriverManger
import time

from selenium.webdriver import ActionChains, Keys
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Set up the actions object
    actions = ActionChains(driver)
    # Open the browser to the URL
    driver.get("https://v1.training-support.net/selenium/dynamic-controls")
    print("Title of the page is: ", driver.title)
    # Find the checkbox and click
    checkbox = driver.find_element(By.ID, "toggleCheckbox")
    checkbox.click()
    time.sleep(1)
    wait = WebDriverWait(driver, 10)
    # Wait till the checkbox disappears
    wait.until(EC.invisibility_of_element_located((By.ID, "dynamicCheckbox")))
    print("Checkbox disappears: ", +checkbox.is_displayed())
    # click the checkbox
    checkbox.click()
    time.sleep(1)
    # Wait till it appears and check the checkbox
    wait.until(EC.visibility_of_element_located((By.ID, "dynamicCheckbox")))
    print("Checkbox appears: ", +checkbox.is_displayed())

File: Selenium/Activities/Activity8.java
package Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity8 {
    public static void main(String[] args) throws InterruptedException {
        // Setup the Firefox driver(GeckoDriver)
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/dynamic-controls");
        System.out.println("Title of the page: " +driver.getTitle());
        // Find the checkbox toggle button and click it
        WebElement checkbox = driver.findElement(By.id("toggleCheckbox"));
        checkbox.click();
        Thread.sleep(1000);
        // Wait till the checkbox disappears
        WebDriverWait wait = new WebDriverWait (driver, Duration.ofSeconds(20));
        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.id("dynamicCheckbox")));
        System.out.println("Checkbox disappears: " +checkbox.isDisplayed());
        // Click the checkbox
        checkbox.click();
        Thread.sleep(1000);
        // Wait till it appears and check the checkbox
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("dynamicCheckbox")));
        System.out.println("Checkbox appears: " +checkbox.isDisplayed());
        driver.quit();
    }
}